orca-retrofit/src/main/groovy/com/netflix/spinnaker/orca/retrofit/RetrofitConfiguration.groovy

```diff
@@ -52,22 +56,41 @@ class RetrofitConfiguration {
    @Value('${okHttpClient.retryOnConnectionFailure:true}')
    boolean retryOnConnectionFailure
 
-   @Bean(name = ["retrofitClient", "okClient"])
-   @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
-   OkClient retrofitClient(@Qualifier("okHttpClientConfiguration") OkHttpClientConfiguration okHttpClientConfig) {
+   @Bean(name = ["retrofitClient"])
+   @Scope(ConfigurableBeanFactory.SCOPE_SINGLETON)
```

Comment:
Was SINGLETON intentional instead of PROTOTYPE? I don't totally understand why we'd have the old OkClient as a PROTOTYPE to begin with, but switching to a SINGLETON is a pretty dramatic departure from prior functionality.

Reply:
I know, seems a bit scary, eh? But I guess that's the right thing to do if you believe the okhttp3 docs:

```
There is no longer a global singleton connection pool. In OkHttp 2.x, all OkHttpClient instances shared a common connection pool by default. In OkHttp 3.x, each new OkHttpClient gets its own private connection pool. Applications should avoid creating many connection pools as doing so prevents connection reuse. Each connection pool holds its own set of connections alive so applications that have many pools also risk exhausting memory!

The best practice in OkHttp 3 is to create a single OkHttpClient instance and share it throughout the application. Requests that needs a customized client should call OkHttpClient.newBuilder() on that shared instance. This allows customization without the drawbacks of separate connection pools.
```

Reply:
Wonderful. That's exactly the sort of behavior I would want out of a connection pool anyway! Thanks.